tools/distpartitioning/data_shuffle.py

```diff
@@ -624,16 +666,17 @@ def gen_dist_partitions(rank, world_size, params):
     memory_snapshot("DataShuffleComplete: ", rank)
 
     #sort node_data by ntype
-    idx = node_data[constants.NTYPE_ID].argsort()
-    for k, v in node_data.items():
-        node_data[k] = v[idx]
-    idx = None
+    for local_part_id in range(params.num_parts//world_size):
+        idx = node_data[constants.NTYPE_ID+"/"+str(local_part_id)].argsort()
+        for k, v in node_data.items():
+            if k.endswith(str(local_part_id)):
```

Comment:
We cannot use assert statement in this instance. 

node_data will have keys which ends with '0', '1', '2' etc... which are local_part_ids. Using this for loop we want to sort which ends with same local_part_ids similarly. So when iterating over the items in the node_data dictionary it will have keys which ends with all the local_part_ids other than the ones which we are targeting for.